internal/integration/entity_synthesis.go

```diff
@@ -0,0 +1,169 @@
+// Package integration ...
+// Copyright 2021 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+package integration
+
+import (
+	"strings"
+
+	sdk_metadata "github.com/newrelic/infra-integrations-sdk/v4/data/metadata"
+	"github.com/newrelic/nri-prometheus/internal/pkg/labels"
+)
+
+// Synthesizer group of rules to synthesis entities
+type Synthesizer struct {
+	rulesByConditions map[Condition]EntityRule
```

Comment:
It would seem that this map is used as a way to keep a list of `Condition`s and the `Rule`s the would apply if matched. This makes sense but I wonder if, rather than a map, we could use a list instead:
```
[]struct{
  cond Condition
  rule Rule
}
```
Lists are cheaper to traverse, grow, and access; and I think it shows better the intent of the data :)

Reply:
I'd also add a small comment here explaining what this is for.

Reply:
you're right, this is probably something that i left for a first implementation where i needed to look for the rules, I will change it for the list approach